openraft/src/core/admin.rs

```diff
@@ -204,6 +209,73 @@ impl<'a, C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> LeaderS
 
         tracing::debug!(?new_config, "new_config");
 
+        if let Err(e) = self.is_all_new_members_line_rate(new_members.cloned().collect::<BTreeSet<_>>(), blocking) {
+            let _ = tx.send(Err(e));
+            return Ok(());
+        }
+
+        self.append_membership_log(new_config, Some(tx)).await?;
+        Ok(())
+    }
+
+    #[tracing::instrument(level = "debug", skip(self, tx))]
+    pub(super) async fn remove_members(
+        &mut self,
+        remove_members: BTreeSet<C::NodeId>,
+        blocking: bool,
+        turn_to_learner: bool,
+        tx: RaftRespTx<ClientWriteResponse<C>, ClientWriteError<C>>,
+    ) -> Result<(), StorageError<C>> {
+        let new_members = {
+            let curr = self.core.effective_membership.membership.clone();
+            let res = curr.remove_members(remove_members, turn_to_learner);
+            let curr = match res {
+                Ok(c) => c,
+                Err(e) => {
+                    let change_err = ChangeMembershipError::MissingNodeInfo(e);
+                    let _ = tx.send(Err(ClientWriteError::ChangeMembershipError(change_err)));
+                    return Ok(());
+                }
+            };
+            curr.get_configs().last().cloned().unwrap()
+        };
+
+        self.change_membership(new_members, blocking, false, tx).await?;
```

Comment:
You do not need to call `curr.remote_members`. Just removing node ids from the last config is enough:
```rust
let new_members = self.core.effective_membership.membership.get_configs().last().cloned().difference(remove_members).collect()
```